dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/path/ArchiveFileIndexInputStreamStrategy.java

```diff
@@ -29,10 +29,19 @@ class ArchiveFileIndexInputStreamStrategy implements FileIndexXsdInputStreamStra
    * SIARDDKPathImportStrategy)
    */
   @Override
-  public InputStream getInputStream(SIARDDKPathImportStrategy siarddkPathImportStrategy) throws ModuleException {
-    ReadStrategy readStrategy = siarddkPathImportStrategy.getReadStrategy();
-    SIARDArchiveContainer mainFolder = siarddkPathImportStrategy.getMainFolder();
-    String xsdFilePath = siarddkPathImportStrategy.getXsdFilePath(SIARDDKConstants.FILE_INDEX);
+  public InputStream getInputStream(SIARDDK1007PathImportStrategy siarddk1007PathImportStrategy) throws ModuleException {
+    ReadStrategy readStrategy = siarddk1007PathImportStrategy.getReadStrategy();
+    SIARDArchiveContainer mainFolder = siarddk1007PathImportStrategy.getMainFolder();
+    String xsdFilePath = siarddk1007PathImportStrategy.getXsdFilePath(SIARDDKConstants.FILE_INDEX);
+
+    return readStrategy.createInputStream(mainFolder, xsdFilePath);
+  }
+
+  @Override
+  public InputStream getInputStream(SIARDDK128PathImportStrategy siarddk128PathImportStrategy) throws ModuleException {
+    ReadStrategy readStrategy = siarddk128PathImportStrategy.getReadStrategy();
+    SIARDArchiveContainer mainFolder = siarddk128PathImportStrategy.getMainFolder();
+    String xsdFilePath = siarddk128PathImportStrategy.getXsdFilePath(SIARDDKConstants.FILE_INDEX);
```

Comment:
After you change the code so that you don't have duplicate code, here you can possibly use polymorphism and continue with just one method.

Reply:
done